theano/gof/cmodule.py

```diff
@@ -415,9 +415,13 @@ def get_module_hash(src_code, key):
             # libraries to link against.
             to_hash += list(key_element)
         elif isinstance(key_element, string_types):
-            if key_element.startswith('md5:'):
-                # This is the md5 hash of the config options. We can stop
-                # here.
+            if (key_element.startswith('md5:') or
+                    key_element.startswith('sha256:') or
+                    key_element.startswith('hash:')):
+                # This is actually a sha256 hash of the config options.
+                # Currently, we still keep md5 to don't break old Theano.
+                # We add 'sha256:' and 'hash:' so that when we change it in
+                # the futur, it won't break this version of Theano.
```

Comment:
I will admit that I don't understand why we need to rifle through the key to find the hash.  Isn't it at a defined position where we could just grab it?

Also, I think we should just add support for `hash:` instead of listing the algorithm used.  That way we can swap out in the future and not have to update this code.